src/models/stores/sorted-documents.ts

```diff
@@ -231,6 +229,69 @@ export class SortedDocuments {
     });
   }
 
+  async updateMetaDataDocs (filter: string, unit: string, investigation: number, problem: number) {
+    const db = this.db.firestore;
+    const converter = typeConverter<IDocumentMetadata>();
+    let query = db.collection("documents").withConverter(converter).where("context_id", "==", this.user.classHash);
+
+    if (filter !== "All") {
+      query = query.where("unit" , "==", unit);
+    }
+    if (filter === "Investigation" || filter === "Problem") {
+      query = query.where("investigation", "==", String(investigation));
+    }
+    if (filter === "Problem") {
+      query = query.where("problem", "==", String(problem));
+    }
+    const queryForUnitNull = db.collection("documents").withConverter(converter)
+                                                       .where("context_id", "==", this.user.classHash)
+                                                       .where("unit" , "==", null);
+    const [docsWithUnit, docsWithoutUnit] = await Promise.all([query.get(), queryForUnitNull.get()]);
+    const docsArray: IDocumentMetadata[] = [];
+
+    const matchedDocKeys = new Set<string>();
+    docsWithUnit.docs.forEach(doc => {
+      if (matchedDocKeys.has(doc.data().key)) return;
+      docsArray.push(doc.data());
+      matchedDocKeys.add(doc.data().key);
+    });
+    docsWithoutUnit.docs.forEach(doc => {
+      if (matchedDocKeys.has(doc.data().key)) return;
+      docsArray.push(doc.data());
+      matchedDocKeys.add(doc.data().key);
+    });
+
+    runInAction(() => {
+      this.firestoreMetadataDocs.replace(docsArray);
+    });
+  }
+
+  async fetchFullDocument(docKey: string) {
+    const metadataDoc = this.firestoreMetadataDocs.find(doc => doc.key === docKey);
+    if (!metadataDoc) return;
+
+    const problemOrdinal = metadataDoc?.type === "problem"
+                             ? `${metadataDoc?.investigation}.${metadataDoc?.problem}`
+                             : undefined;
+    const unit = metadataDoc?.type === "problem" ? metadataDoc?.unit : undefined;
```

Comment:
Rather than looking at the type of the document. I'd base this logic on whether the metadata document has a unit. Any document that has a unit should also have problem and investigation. Planning documents can have units too, and published problem and planning documents can have units. I'm not sure any of those types should actually be shown in the sorted documents but it seems best to keep this logic type agnostic.